### DISCUSSION TITLE
Convert to .avif output broken files

### DISCUSSION THREAD

**Discussion by qvoiriot** [ImageMagick/ImageMagick#4414] [Help]
Hello ,

**Specs :** 
- mac Os Big Sur v 11.6
- brew install aom/libheif/imagemagick on the latest version

I'm using a basic bash script to convert all my {png,jpg,gif} assets to .avif files.

`#!/bin/bash

quality=40
for image in img/**/*.{png,jpg,gif}
do
if [ "${image%.*}.avif" -ot "$image" ]
then
    magick convert -quality $quality "$image" "${image%.*}.avif"
    else
    echo "image $image already existing in .avif format"
fi
done`

it's creating all the .avif files, but they are all broken format when i try to open them. And also files size seems to be weird.
 
A few lines with the verbose:

`img/icons/logo-desktop.png PNG 507x41 507x41+0+0 8-bit sRGB 13792B 0.000u 0:00.002
img/icons/logo-desktop.png=>img/icons/logo-desktop.avif PNG 507x41 507x41+0+0 8-bit sRGB 646B 0.030u 0:00.026

img/backgrounds/QS_2020_S1_ORIGINALS_Carousel.jpg JPEG 1920x800 1920x800+0+0 8-bit sRGB 144085B 0.030u 0:00.026
img/backgrounds/QS_2020_S1_ORIGINALS_Carousel.jpg=>img/backgrounds/QS_2020_S1_ORIGINALS_Carousel.avif JPEG 1920x800 1920x800+0+0 8-bit sRGB 422B 0.040u 0:00.050`

I've also tried to convert to .av1 and that part is working just fine!

I dont get what i can be missing here ? I've read like almost all the opened/closed discussions about the subject and i'm blocked.

If anyone has any clue !

thanks a lot

**Comment by dlemstra** [comment]
Did you check this issue: #4380? Your output file is a PNG image so I suspect you don't have `AVIF` support.

**Reply by qvoiriot** [reply]
thanks @dlemstra  for your answer!
yep i did check it (the issue) and correct me if i'm wrong but i should have AVIF support as libheif aom and all dependencies are installed ?
otherwise how can i check that ?

**Reply by dlemstra** [reply]
You will also need to have ImageMagick linked with that version of libheif.

**Reply by qvoiriot** [reply]
thanks @dlemstra  - how can i see if it's linked then ?

**Reply by dlemstra** [reply]
By looking at your output it looks like you don't have support for AVIF.

**Reply by qvoiriot** [reply]
Well seems that i manage to fix this by :

- brew reinstall --build-from-source imagemagick
- brew reinstall --build-from-source libheif

Everything is now working fine 👍
